lib/producer.js

```diff
@@ -119,7 +119,9 @@ class Producer extends EventEmitter {
             return Promise
                     .fromCallback((cb) => socket.emit('end', { time, code }, cb))
                     .tap(() => {
-                      socket.close();
+                      if (!_.isNil(socket)) {
+                        socket.close();
+                      }
                     });
```

Comment:
this is not needed - would have failed on socket.emit above it -- remove this check